pydle/client.py

```diff
@@ -100,7 +100,12 @@ def connect(self, hostname=None, port=None, reconnect=False, eventloop=None, **k
         # Reset attributes and connect.
         if not reconnect:
             self._reset_connection_attributes()
-        self._connect(hostname=hostname, port=port, reconnect=reconnect, **kwargs)
+        try:
+            self._connect(hostname=hostname, port=port, reconnect=reconnect, **kwargs)
+        except OSError:
+            self.on_disconnect(
+                expected=False,
```

Comment:
May you please just take the diff and make the patch.
This is an important fix and it should be merged ASAP.
Would you like a new PR?